In [111]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [112]:
import matplotlib.pyplot as plt

In [113]:
%matplotlib widget

In [114]:
import pandas as pd
from IPython.display import display
pd.options.display.max_columns = None

# Checking for bugs

In [115]:
name_of_evil_file = "20180919_FC_LFP2_cen14_01_cc_01.res"
datapath = "/Users/jepe/scripting/cellpy/dev_data/bugfixing"

In [116]:
import os
import sys
import time
from pathlib import Path
print(f"running {sys.argv[0]}")

import cellpy
from cellpy import log
from cellpy import cellreader
from cellpy.parameters import prms

prms.Reader.use_cellpy_stat_file = False
#prms.Reader.cycle_mode = "cathode"
prms.Reader.sorted_data = False
log.setup_logging(default_level="INFO", custom_log_dir=os.getcwd())

filename = Path(datapath) / name_of_evil_file

assert os.path.isfile(filename)

running /Users/jepe/miniconda3/envs/cellpy/lib/python3.6/site-packages/ipykernel_launcher.py
(cellpy) - Filename: cellpy_errors.log
(cellpy) - Full path: /Users/jepe/scripting/cellpy/dev_utils/cellpy_errors.log
(cellpy) - Filename: cellpy_info.log
(cellpy) - Full path: /Users/jepe/scripting/cellpy/dev_utils/cellpy_info.log
(cellpy) - Filename: cellpy_debug.log
(cellpy) - Full path: /Users/jepe/scripting/cellpy/dev_utils/cellpy_debug.log


In [117]:
d = cellreader.CellpyData()
d.from_raw(filename)
d.set_mass(0.12)
d.make_step_table()
d.make_summary()

# checking extracting cycles
n = d.get_number_of_cycles()
c = d.get_cycle_numbers()

(cellpy) - created CellpyData instance


/Users/jepe/scripting/cellpy/cellpy/readers/cellreader.py:2960: UserWarning: Cycle 11 is missing!
  if j + 1 not in unique_steps:
/Users/jepe/scripting/cellpy/cellpy/readers/cellreader.py:2960: UserWarning: Cycle 22 is missing!
  if j + 1 not in unique_steps:
/Users/jepe/scripting/cellpy/cellpy/readers/cellreader.py:2960: UserWarning: Cycle 33 is missing!
  if j + 1 not in unique_steps:
/Users/jepe/scripting/cellpy/cellpy/readers/cellreader.py:2960: UserWarning: Cycle 44 is missing!
  if j + 1 not in unique_steps:
/Users/jepe/scripting/cellpy/cellpy/readers/cellreader.py:2960: UserWarning: Cycle 55 is missing!
  if j + 1 not in unique_steps:
/Users/jepe/scripting/cellpy/cellpy/readers/cellreader.py:2960: UserWarning: Cycle 66 is missing!
  if j + 1 not in unique_steps:
/Users/jepe/scripting/cellpy/cellpy/readers/cellreader.py:2960: UserWarning: Cycle 77 is missing!
  if j + 1 not in unique_steps:
/Users/jepe/scripting/cellpy/cellpy/readers/cellreader.py:2960: UserWarning: Cycle 88 is m

In [118]:

current = d.dataset.dfdata
steps = d.dataset.step_table

In [119]:
steps.loc[(steps.step==9) | (steps.step==2), (steps.columns.str.startswith("cha")|steps.columns.str.startswith("ty"))]

charge_avr  charge_std    charge_min  charge_max  charge_first  \
1      0.000535    0.000571  1.321240e-10    0.001715  1.321240e-10   
7      0.000402    0.000419  0.000000e+00    0.001238  0.000000e+00   
13     0.000412    0.000395  6.302670e-10    0.001159  6.302670e-10   
19     0.000406    0.000389  7.929460e-10    0.001151  7.929460e-10   
25     0.000411    0.000391  6.302830e-10    0.001142  6.302830e-10   
31     0.000403    0.000385  0.000000e+00    0.001140  0.000000e+00   
37     0.000401    0.000384  0.000000e+00    0.001135  0.000000e+00   
43     0.000402    0.000385  4.676310e-10    0.001130  4.676310e-10   
49     0.000403    0.000385  0.000000e+00    0.001125  0.000000e+00   
55     0.000403    0.000385  0.000000e+00    0.001123  0.000000e+00   
61     0.000609    0.000389  7.160638e-09    0.001278  7.160638e-09   
67     0.000388    0.000407  0.000000e+00    0.001199  0.000000e+00   
73     0.000396    0.000382  3.049810e-10    0.001118  3.049810e-10   
79     0.000386    0.000375  0.000000e+00    0.001109  0.000000e+00   
85     0.000387    0.000376  8.742370e-10    0.001107  8.742370e-10   
91     0.000385    0.000375  0.000000e+00    0.001101  0.000000e+00   
97     0.000383    0.000374  6.099580e-10    0.001101  6.099580e-10   
103    0.000383    0.000374  4.066430e-10    0.001097  4.066430e-10   
109    0.000383    0.000374  4.066370e-10    0.001091  4.066370e-10   
115    0.000383    0.000374  6.302880e-10    0.001089  6.302880e-10   
121    0.000589    0.000381  6.607139e-09    0.001247  6.607139e-09   
127    0.000370    0.000397  0.000000e+00    0.001166  0.000000e+00   
133    0.000376    0.000371  0.000000e+00    0.001080  0.000000e+00   
139    0.000368    0.000365  0.000000e+00    0.001073  0.000000e+00   
145    0.000369    0.000365  4.472980e-10    0.001069  4.472980e-10   
151    0.000369    0.000365  0.000000e+00    0.001065  0.000000e+00   
157    0.000371    0.000366  0.000000e+00    0.001063  0.000000e+00   
163    0.000371    0.000366  0.000000e+00    0.001059  0.000000e+00   
169    0.000367    0.000364  5.286190e-10    0.001052  5.286190e-10   
175    0.000365    0.000360  3.456400e-10    0.001049  3.456400e-10   
..          ...         ...           ...         ...           ...   
361    0.000508    0.000342  2.495605e-09    0.001102  2.495605e-09   
367    0.000302    0.000340  5.896300e-10    0.001010  5.896300e-10   
373    0.000306    0.000315  4.676340e-10    0.000912  4.676340e-10   
379    0.000327    0.000314  0.000000e+00    0.000905  0.000000e+00   
385    0.000299    0.000309  0.000000e+00    0.000899  0.000000e+00   
391    0.000300    0.000309  1.118260e-09    0.000895  1.118260e-09   
397    0.000294    0.000303  1.158989e-09    0.000889  1.158989e-09   
403    0.000289    0.000301  6.912770e-10    0.000884  6.912770e-10   
409    0.000289    0.000301  6.912750e-10    0.000878  6.912750e-10   
415    0.000290    0.000301  0.000000e+00    0.000875  0.000000e+00   
421    0.000489    0.000331  2.281863e-09    0.001063  2.281863e-09   
427    0.000287    0.000327  0.000000e+00    0.000970  0.000000e+00   
433    0.000288    0.000300  0.000000e+00    0.000869  0.000000e+00   
439    0.000289    0.000300  0.000000e+00    0.000861  0.000000e+00   
445    0.000279    0.000293  0.000000e+00    0.000855  0.000000e+00   
451    0.000287    0.000297  0.000000e+00    0.000850  0.000000e+00   
457    0.000275    0.000289  6.099540e-10    0.000846  6.099540e-10   
463    0.000277    0.000289  0.000000e+00    0.000842  0.000000e+00   
469    0.000276    0.000289  8.742660e-10    0.000837  8.742660e-10   
475    0.000277    0.000292  7.929470e-10    0.000847  7.929470e-10   
481    0.000476    0.000325  2.129500e-09    0.001041  2.129500e-09   
487    0.000276    0.000320  0.000000e+00    0.000936  0.000000e+00   
493    0.000274    0.000290  0.000000e+00    0.000842  0.000000e+00   
499    0.000267    0.000284  0.000000e+00    0.000833  0.000000e+00   
505    0.000267    0.000284  

In [120]:
steps.loc[(steps.cycle==2) & (steps.step==9)].T

7
cycle                      2
step                       9
sub_step                   1
point_avr                905
point_std            23.5266
point_min                865
point_max                945
point_first              865
point_last               945
point_delta          9.24855
step_time_avr        1976.87
step_time_std        2058.67
step_time_min         0.0045
step_time_max        6085.92
step_time_first       0.0045
step_time_last       6085.92
step_time_delta  1.35243e+08
current_avr      0.000732255
current_std       1.2277e-07
current_min      0.000731923
current_max      0.000732323
current_first    0.000731923
current_last     0.000732311
current_delta      0.0529331
voltage_avr          3.17765
voltage_std         0.160558
voltage_min          2.85084
voltage_max              3.4
voltage_first        2.85084
voltage_last             3.4
voltage_delta         19.263
charge_avr       0.000402134
charge_std       0.000418774
charge_min                 0
charge_max          0.001238
charge_first               0
charge_last         0.001238
charge_delta               1
discharge_avr              0
discharge_std              0
discharge_min              0
discharge_max              0
discharge_first            0
discharge_last             0
discharge_delta            0
ir_avr               32.8247
ir_std                     0
ir_min               32.8247
ir_max               32.8247
ir_first             32.8247
ir_last              32.8247
ir_delta                   0
type                  charge
sub_type                None
info

charge_avr	0.000411908
charge_std	0.000394592
charge_min	6.30267e-10
charge_max	0.00115869
charge_first	6.30267e-10
charge_last	0.00115869
charge_delta	1.83841e+08



charge_avr	0.000402134
charge_std	0.000418774
charge_min	0
charge_max	0.001238
charge_first	0
charge_last	0.001238
charge_delta	
 
 
 

In [121]:
fig, (ax1, ax2) = plt.subplots(2, sharex=True)
ax1.plot(current.Charge_Capacity, label="current")
ax1.plot(normal.Charge_Capacity, label="before", alpha=0.3)
ax1.legend()
ax2.plot(current.Step_Index)

FigureCanvasNbAgg()

In [122]:

c_current = current.loc[(current.Cycle_Index == 2)]

In [123]:
fig, ax = plt.subplots()
ax.plot(c_current.Voltage, '.r', label="current")
ax.legend()

FigureCanvasNbAgg()

In [7]:
dataset = d.dataset

In [124]:
normal = dataset.dfdata.copy()
steps = dataset.step_table.copy()
normal2 = normal.sort_values("Data_Point").reset_index()

In [125]:
c2_filter = (normal.Cycle_Index == 2) &  (normal.Step_Index == 9)
c3_filter = (normal.Cycle_Index == 3) &  (normal.Step_Index == 9)
c2 = normal.loc[c2_filter, ["Data_Point", "Step_Time", "Charge_Capacity"]]
c3 = normal.loc[c3_filter, ["Data_Point", "Step_Time", "Charge_Capacity"]]
c2b_filter = (normal2.Cycle_Index == 2) &  (normal2.Step_Index == 9)
c3b_filter = (normal2.Cycle_Index == 3) &  (normal2.Step_Index == 9)
c2b = normal2.loc[c2b_filter, ["Data_Point", "Step_Time", "Charge_Capacity"]]
c3b = normal2.loc[c3b_filter, ["Data_Point","Step_Time", "Charge_Capacity"]]

In [126]:
fig, (ax1, ax2) = plt.subplots(2,1)
ax1.plot(c2.Step_Time, c2.Charge_Capacity, '-*')
ax1.plot(c2b.Step_Time, c2b.Charge_Capacity, '-ro')
ax2.plot(c2.Charge_Capacity)
ax2.plot(c2b.Charge_Capacity, 'r')
ax2.plot(c3b.Data_Point, c3b.Charge_Capacity, 'g')

FigureCanvasNbAgg()

In [132]:
v,c = d.get_cap()
#print(caps)
# checking creating dqdv

#d.to_csv(datapath, cycles=True)

In [137]:
fig, ax = plt.subplots(1)
ax.plot(v,c)

FigureCanvasNbAgg()